src/lib/dnssd/ResolverProxy.h

```diff
@@ -71,16 +89,35 @@ class ResolverProxy : public Resolver
         return mDelegate != nullptr ? CHIP_NO_ERROR : CHIP_ERROR_NO_MEMORY;
     }
 
-    void SetResolverDelegate(ResolverDelegate * delegate) override
+    void SetOperationalDelegate(OperationalResolveDelegate * delegate) override
     {
-        VerifyOrReturn(mDelegate != nullptr);
-        mDelegate->SetDelegate(delegate);
+        if (mDelegate != nullptr)
+        {
+            mDelegate->SetOperationalDelegate(delegate);
+        }
+        else
+        {
+            ChipLogError(Discovery, "Failed to proxy operational discovery: missing delegate");
```

Comment:
Isn't this more like "not initialized"?  mDelegate gets set up in `Init`.  Same thing in SetCommissioningDelegate

Reply:
If you get here, is because init was not called or it was with delegate nullptr and error was ignored.
I figured we need some message, and this is clear enough about what is bad (I need to use a delegate and delegate is null).